test/v3/controllers/SubmitForeignPropertyBsasControllerSpec.scala

```diff
@@ -172,7 +176,7 @@ class SubmitForeignPropertyBsasControllerSpec
           .wrap((), SubmitForeignPropertyBsasHateoasData(nino, bsasId))
           .returns(HateoasWrapper((), Seq(testHateoasLink)))
 
-        val result: Future[Result] = controller.handleRequest(nino, bsasId)(fakePostRequest(requestJson))
+        val result: Future[Result] = controller.handleRequest(nino, bsasId, Some(rawTaxYear))(fakePostRequest(requestJson))
```

Comment:
Should we add a pre-TYS tests here as well, or can we just replace all the non-TYS tests with TYS? It looks like in previous endpoints we have been just adding an extra-TYS request test in addition to these ones

Reply:
I don't think so. The controller has no logic that depends on the taxYear: it just passes it to the parser. So as long as the parser gets the right value (checked by the mock) we are ok.